lib/std/fs/path.zig

```diff
@@ -1182,3 +1182,69 @@ fn testRelativeWindows(from: []const u8, to: []const u8, expected_output: []cons
     defer testing.allocator.free(result);
     testing.expectEqualSlices(u8, expected_output, result);
 }
+
+/// Returns the extension of the file name (if any).
+/// This function will search for the file extension (separated by a `.`) and will return the text after the `.`.
+/// Files that end with `.` are considered to have no extension, files that start with `.`
+/// Examples:
+/// - `"main.zig"`     ⇒ `"zig"`
+/// - `"src/main.zig"` ⇒ `"zig"`
+/// - `".gitignore"`   ⇒ `null`
+/// - `"keep."`        ⇒ `null`
+/// - `"src.keep.me"`  ⇒ `"me"`
+pub fn extension(path: []const u8) ?[]const u8 {
+    const filename = basename(path);
+    return if (std.mem.lastIndexOf(u8, filename, ".")) |index|
+        if (index == 0 or index == filename.len - 1)
```

Comment:
No need to compare the first and last character.

```suggestion
    return if (std.mem.lastIndexOf(u8, filename[1..filename.len-1], ".")) |index|
        if (index == 0)
```

Reply:
This will introduce another check, and will probably be less understandable:
```zig
return if (filename.len > 2 and std.mem.lastIndexOf(u8, filename[1..filename.len-1], ".")) |index|
```
doesn't convey the idea that we actively ignore the first and the last char if they are `.`